python/src/main/python/drivers/graphicsfuzz_piglit_converter.py

```diff
@@ -0,0 +1,191 @@
+    test_header = TEST_HEADER + '\n'
+    uniforms_in_shader = get_uniforms_from_shader(shader_line_list)
+    for dec in uniforms_in_shader:
+        dec = dec.strip(';')
+        test_header += dec + ' '
+        variable_name = dec.split(' ')[-1]
+        if not variable_name in shader_job_json_parsed:
+            raise IOError('Malformed shader job - ' + dec + ' not found in job.')
+        uniform_args = shader_job_json_parsed.get(variable_name).get('args')
+        test_header += str(uniform_args[0:]).strip('[]').replace(',', '') + '\n'
+    test_header += DRAW_COMMAND
+    return test_header
+
+# Returns a list of all uniform declarations in a shader.
+def get_uniforms_from_shader(shader_lines: List) -> List:
+    return (line for line in shader_lines if is_uniform(line))
+
+# Helper function to see if a given string is a GLSL uniform declaration.
+def is_uniform(line: str) -> bool:
+    return ('uniform' in line)
+
+# Returns the first occurrence of a GLSL '#version' string in a list of lines of shader code.
+def get_version_header_from_shader(shader_lines: List) -> str:
+    return next(line for line in shader_lines if is_version_header(line))
+    
+# Helper function to see if a given string is a GLSL preprocessor version string.
+def is_version_header(line: str) -> bool:
+    return (SHADER_VERSION_FLAG in line)
+    
+# Helper function to parse a shader job JSON file into a list of properties.
+# Throws IOError if the file can't be parsed.
+def get_json_properties(shader_job: str) -> List:
+    with open(shader_job, 'r', encoding = 'utf-8', errors = 'ignore') as job:
+        json_parsed = json.load(job)
+    if not json_parsed:
+        raise IOError('Malformed shader job file.')
+    return json_parsed
+
+# Checks to see if a shader job JSON and a corresponding shader file exist at
+# the same location as the shader job. Throws FileNotFoundError if either cannot be found.
+def check_shader_job_exists(shader_job_name: str) -> None:
+    if not os.path.isfile(shader_job_name):
+        raise FileNotFoundError('Input shader job ' + shader_job_name + ' not found.')
+    # We need to make sure a corresponding shader exists as well.
+    shader_name = get_shader_from_job(shader_job_name)
+    # TODO: Figure out how piglit handles compute shaders so we can support them here.
+    if not os.path.isfile(shader_name):
+        raise FileNotFoundError('Fragment shader ' + shader_name + ' not found.')
+
+# Helper function to get the filename of a shader file from its corresponding shader job.
+def get_shader_from_job(shader_job: str) -> str:
+    return shader_job.split(".")[0] + ".frag"
+
+# Helper function to get the filename of the new shader test from the given shader job.
+def get_shader_test_from_job(shader_job: str) -> str:
+    return shader_job.split(".")[0] + ".shader_test"
+
+def main_helper(args: List[str]) -> int:
+    description = (
+        'Given a GraphicsFuzz shader job JSON file, produce a Mesa piglit \
```

Comment:
Nit: I believe you can just close the string `'` and re-open it on the next line `'` instead of using `\`, as long as you have parentheses.

Reply:
Thanks, I wasn't aware of that.